dll/ntdll/ldr/ldrutils.c

```diff
@@ -2455,6 +2455,7 @@ LdrpLoadDll(IN BOOLEAN Redirected,
     RtlCopyUnicodeString(&RawDllName, DllName);
 
     /* Find the extension, if present */
+    /* NOTE: Access violation is expected here in some cases (Buffer[-1]) */
```

Comment:
What code here would cause an access violation? If there is some, we should fix it instead of adding a comment, but I don't see any.
`p` may end up pointing to `&Buffer[-1]` here, but since the while loop has a `p >= Buffer` check, it doesn't seem to ever get dereferenced in that case, so this is safe as far as I can tell. Am I missing something?

Reply:
`RtlCopyUnicodeString` can set `Buffer = NULL` and `Length = 0`. This is the case.

Reply:
```txt
#include <windows.h>
#include <stdio.h>

LONG CALLBACK exception_handler(EXCEPTION_POINTERS *ExceptionInfo)
{
    puts("You are an idiot");
    return 0;
}
 
int main()
{
    LPWSTR Buffer = NULL;
    USHORT Length = 0;
    WCHAR c, *p = Buffer + Length - 1;
    SetUnhandledExceptionFilter(exception_handler);
    while (p >= Buffer)
    {
        c = *p--;
        if (c == L'.' || c == L'\\')
            break;
    }
    puts("OK");
}
```

![image](https://github.com/reactos/reactos/assets/2107452/aac69036-f610-4e5e-9ccd-8209488a494a)